README.md

```diff
@@ -375,6 +375,54 @@ interface CacheProfile
 }
 ```
 
+### Flexible caching with stale-while-revalidate (SWR)
+
+The package supports flexible caching with stale-while-revalidate behavior using explicit middleware. With SWR, cached responses have two time periods:
+
+- **Fresh period**: The cache is considered fresh and served immediately without revalidation
+- **Stale period**: The cache is served immediately but triggers a background refresh
+
+This means users always get instant responses, even when cache needs updating.
+
+#### Using flexible caching
+
+Configure flexible caching per route using the `flexible()` method:
+
+```php
+use Spatie\ResponseCache\Middlewares\CacheResponse;
+
+// Fresh for 3 minutes, then stale for 15 minutes
+Route::get('/api/posts', 'PostController@index')
+    ->middleware(CacheResponse::flexible(180, 900));
+
+// Defer refresh to background during stale period
+Route::get('/api/live-data', 'LiveDataController@index')
+    ->middleware(CacheResponse::flexible(60, 300, true));
+
+// Middleware group, fresh for 5 seconds, stale for 10 seconds
+Route::middleware(CacheResponse::flexible(5, 10))->group(function () {
+    Route::get('/test', function () {
+        return rand();
+    });
+});
+
+// With defer and cache tags
+Route::get('/api/posts', 'PostController@index')
+    ->middleware(CacheResponse::flexible(180, 900, false, 'posts', 'api'));
+```
+
+The `flexible()` method accepts:
```

Comment:
Instead of giving all examples first, and then explaining all parameters, it would be more readable to give a simple example, explain that fully. And then introduce another parameter, give an example for the new parameter and so on.

That will feel a lot more "human" that what we have now.